src/pilot/dell_nfv.py

```diff
@@ -0,0 +1,376 @@
+                    hostos_cpu_count)
+                cmds.append(
+                    "sed -i 's|dellnfv::numa::vcpu_pin_set:.*"
+                    "|dellnfv::numa::vcpu_pin_set: \"" +
+                    vcpu_pin_set +
+                    "\"|' " +
+                    file_path)
+                cmds.append(
+                    'sed -i "s|  # NovaVcpuPinSet|  ' +
+                    'NovaVcpuPinSet|" ' +
+                    file_path)
+                cmds.append(
+                    "sed -i 's|NovaVcpuPinSet:.*|NovaVcpuPinSet: \"" +
+                    vcpu_pin_set +
+                    "\"|' " +
+                    file_path)
+
+            for cmd in cmds:
+                status = os.system(cmd)
+                logger.debug("cmd: {}".format(cmd))
+            if status != 0:
+                raise Exception(
+                    "Failed to execute the command {}"
+                    " with error code {}".format(
+                        cmd, status))
+
+        except Exception as error:
+            message = "Exception {}: {}".format(
+                type(error).__name__, str(error))
+            logger.error("{}".format(message))
+            raise Exception(
+                "Failed to modify the dell_environment.yaml"
+                " at location {}".format(file_path))
+
+    def get_dell_compute_nodes_hostnames(self, nova):
+        try:
+            logger.info("Getting dellnfv compute node hostnames")
+            # Create host object
+            host_obj = hosts.HostManager(nova)
+
+            # Get list of dell nfv nodes
+            dell_hosts = []
+
+            for host in host_obj.list():
+                if "dell-compute" in host.host_name:
+                    hostname = str(host.host_name)
+                    dell_hosts.append(hostname)
+            return dell_hosts
+        except Exception as error:
+            message = "Exception {}: {}".format(
+                type(error).__name__, str(error))
+            logger.error("{}".format(message))
+            raise Exception("Failed to get the Dell Compute nodes.")
+
+    def edit_aggregate_environment_file(
+            self, hostname_list):
+        logger.info("Editing create aggregate environment file")
+        file_path = home_dir \
+            + '/pilot/templates/create_aggregate_environment.yaml'
+        if not os.path.isfile(file_path):
+            raise Exception(
+                "The create_aggregate_environment.yaml file does not exist")
+        cmd = (
+            'sed -i "s|hosts:.*|hosts: ' +
+            str(hostname_list) +
+            '|" ' +
+            file_path)
+
+        status = os.system(cmd)
+        logger.info("cmd: {}".format(cmd))
+        if status != 0:
+            raise Exception(
+                "Failed to execute the command {}"
+                " with error code {}".format(
+                    cmd, status))
+
+    def create_aggregate(self):
+        try:
```

Comment:
What is the reason for creating the aggregate using heat?  The prior code that created it using nova directly seems much more straight forward.

Reply:
I'm ok changing the heat method for creating aggregates, but if we change, we need to stick with this new method and not change it again.  We need to work toward stabilization of the code, and changing the way of doing things without a compelling reason does not help move us in that direction.

Reply:
As discussed 
Reverted back to old code.